### PR TITLE
[Refactor] Alter Job Manager Support serialized and deserialized with JSON format

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## Problem Summary:
Fixes # (issue)

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 22 / 64 (34.38%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|0|4|00.00%|[276, 282, 335, 814]|
|:large_blue_circle:|com/starrocks/alter/AlterJobMgr.java|10|43|23.26%|[743, 746, 1081, 1082, 1084, 1085, 1087, 1088, 1089, 1090, 1092, 1093, 1094, 1095, 1097, 1098, 1101, 1103, 1104, 1106, 1107, 1112, 1113, 1114, 1118, 1119, 1121, 1122, 1127, 1128, 1129, 1133, 1135]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|6|11|54.55%|[1413, 3282, 3286, 3290, 3415]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/metric/MetricRepo.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/alter/AlterJobMgr.java [hunk 1]
```diff
@@ -1072,4 +1076,61 @@ public AlterHandler getMaterializedViewHandler() {
...
+
+        writer.close();
+    }
+
+    public void load(DataInputStream dis) throws IOException, SRMetaBlockException, SRMetaBlockEOFException {
+        SRMetaBlockReader reader = new SRMetaBlockReader(dis, AlterJobMgr.class.getName());
+        try {
+            int schemaChangeJobSize = (int) reader.readJson(int.class);
+            for (int i = 0; i != schemaChangeJobSize; ++i) {
+                // 2 json for each user(kv)
```
Comment (gengjun-git) [inline fe/fe-core/src/main/java/com/starrocks/alter/AlterJobMgr.java:1105]: remove this commet?

File: fe/fe-core/src/main/java/com/starrocks/alter/AlterJobMgr.java [hunk 2]
```diff
@@ -1072,4 +1076,61 @@ public AlterHandler getMaterializedViewHandler() {
     public AlterHandler getClusterHandler() {
         return this.clusterHandler;
     }
+
+    public void save(DataOutputStream dos) throws IOException, SRMetaBlockException {
+        Map<Long, AlterJobV2> schemaChangeAlterJobs = schemaChangeHandler.getAlterJobsV2();
+        Map<Long, AlterJobV2> materializedViewAlterJobs = materializedViewHandler.getAlterJobsV2();
+
+        int cnt = 1 + schemaChangeAlterJobs.size() + 1 + materializedViewAlterJobs.size();
```
Comment (gengjun-git) [inline fe/fe-core/src/main/java/com/starrocks/alter/AlterJobMgr.java:1084]: better to write current object for future extension.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 22 / 64 (34.38%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|0|4|00.00%|[278, 284, 337, 818]|
|:large_blue_circle:|com/starrocks/alter/AlterJobMgr.java|10|43|23.26%|[743, 746, 1081, 1082, 1084, 1085, 1087, 1088, 1089, 1090, 1092, 1093, 1094, 1095, 1097, 1098, 1101, 1103, 1104, 1105, 1106, 1111, 1112, 1113, 1117, 1118, 1119, 1120, 1125, 1126, 1127, 1131, 1133]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|6|11|54.55%|[1414, 3281, 3285, 3289, 3414]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/metric/MetricRepo.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 44 / 64 (68.75%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|0|4|00.00%|[278, 284, 337, 818]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|6|11|54.55%|[1414, 3281, 3285, 3289, 3414]|
|:large_blue_circle:|com/starrocks/alter/AlterJobMgr.java|32|43|74.42%|[743, 746, 1094, 1095, 1112, 1113, 1119, 1120, 1125, 1126, 1127]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/metric/MetricRepo.java|1|1|100.00%|[]|

**Review by nshangyiming — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Pull request merged**